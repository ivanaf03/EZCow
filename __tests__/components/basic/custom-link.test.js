import React from "react";

import {render, fireEvent} from "@testing-library/react-native";
import { router } from 'expo-router';

import CustomLink from "../../../components/basic/custom-link";

jest.mock("expo-router", () => ({
    router: {
        push: jest.fn(),
    },
}));

describe("CustomLink", () => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", () => {
        const tree = render(<CustomLink text="Test" to="Test"/>);
        expect(tree).toMatchSnapshot();
    });

    it("should navigate to the correct screen", () => {
        const tree = render(<CustomLink text="Test" to="Test"/>);
        fireEvent.press(tree.getByText("Test"));
        expect(router.push).toHaveBeenCalledWith("Test");
    });

});