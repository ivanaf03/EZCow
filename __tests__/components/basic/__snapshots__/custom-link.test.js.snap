// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLink should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      [
        {
          "color": "#6bc0eb",
        },
        {
          "fontSize": 20,
          "lineHeight": 28,
        },
        {
          "textAlign": "center",
        },
        {
          "fontFamily": "Nunito-Italic",
        },
        {
          "textDecorationLine": "underline",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;
